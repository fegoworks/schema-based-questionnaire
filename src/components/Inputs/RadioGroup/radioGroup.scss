.RadioGroup {
  margin-bottom: 0.6rem;
  display: flex;
  flex-direction: column;
}
.RadioGroup label {
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  font-weight: 500;

  span {
    margin-left: 0.5rem;
  }
}

.RadioGroup input[type="radio"] {
  margin: 0.5rem 0;
}

.RadioGroup .error {
  color: #d44;
  margin: 0;
}

.RadioGroup input:focus-visible,
.RadioGroup input:focus {
  border-radius: 3px;
  outline: 4px auto -webkit-focus-ring-color;
  outline-color: #ddd;
}

input[type="radio"] {
  width: 2em;
  height: 2rem;
  accent-color: green;
}
