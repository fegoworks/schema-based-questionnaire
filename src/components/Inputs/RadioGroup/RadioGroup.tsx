import "./radioGroup.scss";
import { FunctionComponent } from "react";
import { FieldProps } from "../../formField";

const RadioGroup: FunctionComponent<FieldProps> = ({
  error,
  name,
  onBlur,
  onChange,
  touched,
  options,
  disabled,
  value,
}) => {
  return (
    <div className="RadioGroup" role="group">
      {options?.map((option) => (
        <label key={option.value}>
          <input
            onChange={onChange}
            onBlur={onBlur}
            type="radio"
            name={name}
            value={option.value}
            checked={option.value === value}
            disabled={disabled}
          />
          <span className={disabled ? "disabled" : ""}>{option.label}</span>
        </label>
      ))}
      {error && touched && <p className="error">{error}</p>}
    </div>
  );
};

export default RadioGroup;
