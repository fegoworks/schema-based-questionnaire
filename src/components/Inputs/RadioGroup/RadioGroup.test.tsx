import { fireEvent, render } from "@testing-library/react";
import { describe, expect } from "vitest";
import RadioGroup from "./RadioGroup";

const opt = [
  { label: "Bread", value: "bread" },
  { label: "Sandwich", value: "sandwich" },
  { label: "Pancakes", value: "pancakes" },
];
const defaultProps = {
  name: "breakfast",
  options: opt,
  disabled: false,
};

describe("Component - RadioGroup", () => {
  it("should render RadioGroup component", () => {
    const { container } = render(<RadioGroup {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should render a RadioGroup for each option", () => {
    const { getByRole } = render(
      <RadioGroup {...defaultProps} value="Sandwich" />
    );
    expect(getByRole("radio", { name: "Bread" })).toBeInTheDocument();
    expect(getByRole("radio", { name: "Pancakes" })).toBeInTheDocument();
    expect(getByRole("radio", { name: "Sandwich" })).toBeInTheDocument();
  });

  it("should render a RadioGroup for each option", () => {
    const { getByRole } = render(<RadioGroup {...defaultProps} />);
    const TextOption = getByRole("radio", { name: "Bread" });

    expect(TextOption).not.toBeChecked();
    fireEvent.click(getByRole("radio", { name: "Bread" }));
    expect(TextOption).toBeChecked();
  });

  it("should show an error message if validation is incorrect", () => {
    const { getByText } = render(
      <RadioGroup {...defaultProps} error={"Some error"} touched={true} />
    );
    expect(getByText("Some error")).toBeTruthy();
  });
});
