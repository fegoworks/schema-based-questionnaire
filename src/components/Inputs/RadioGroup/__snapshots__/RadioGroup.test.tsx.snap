// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component - RadioGroup > should render RadioGroup component 1`] = `
<div>
  <div
    class="RadioGroup"
    role="group"
  >
    <label>
      <input
        name="breakfast"
        type="radio"
        value="bread"
      />
      <span>
        Bread
      </span>
    </label>
    <label>
      <input
        name="breakfast"
        type="radio"
        value="sandwich"
      />
      <span>
        Sandwich
      </span>
    </label>
    <label>
      <input
        name="breakfast"
        type="radio"
        value="pancakes"
      />
      <span>
        Pancakes
      </span>
    </label>
  </div>
</div>
`;
